modules/detection/include/visp3/detection/vpDetectorAprilTag.h

```diff
@@ -251,6 +251,7 @@ class VISP_EXPORT vpDetectorAprilTag : public vpDetectorBase
               std::vector<vpHomogeneousMatrix> &cMo_vec, std::vector<vpHomogeneousMatrix> *cMo_vec2=NULL,
               std::vector<double> *projErrors=NULL, std::vector<double> *projErrors2=NULL);
 
+  std::vector<std::vector<vpPoint> > getPoint3D();
```

Comment:
Current `getPoint3D()` will not work if `getPose()` is used.
Also, it requires to actually compute the tags pose while the 3D tag coordinates can be computed easily.

I suggest to add `getPoint3D()` independant of the pose computation machinery in `vpDetectorAprilTagDetector`.
This would allow using `getTagsCorners()` introduced in #680 to compute the pose with OpenCV or any other library.

Reply:
You are right. I will do the changes